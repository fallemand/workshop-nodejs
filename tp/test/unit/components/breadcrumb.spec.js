const React = require('react');
const renderer = require('react-test-renderer');
const Breadcrumb = require('../../../views/components/Breadcrumb');

describe('Breadcrumb test using snapshots', () => {
  const response = {
    data: {
      categories: ['ComputaciÃ³n', 'Apple', 'Macbook'],
    },
  };

  it('Render categories', () => {
    const component = renderer.create(<Breadcrumb categories={response.data.categories} />);
    const json = component.toJSON();

    expect(json).toMatchSnapshot();
  });

  it('Render categories with query', () => {
    const component = renderer.create(<Breadcrumb categories={response.data.categories} query="macb" />);
    const json = component.toJSON();

    expect(json).toMatchSnapshot();
  });
});
