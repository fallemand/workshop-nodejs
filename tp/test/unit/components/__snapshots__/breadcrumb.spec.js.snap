// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb test using snapshots Render categories 1`] = `
<ul
  className="breadcrumb"
>
  <li
    className="breadcrumb__element"
  >
    <a
      href="/app/"
    >
      Inicio
    </a>
  </li>
  <li
    className="breadcrumb__element"
  >
    Computación
  </li>
  <li
    className="breadcrumb__element"
  >
    Apple
  </li>
  <li
    className="breadcrumb__element"
  >
    Macbook
  </li>
</ul>
`;

exports[`Breadcrumb test using snapshots Render categories with query 1`] = `
<ul
  className="breadcrumb"
>
  <li
    className="breadcrumb__element"
  >
    <a
      href="/app/"
    >
      Inicio
    </a>
  </li>
  <li
    className="breadcrumb__element"
  >
    Computación
  </li>
  <li
    className="breadcrumb__element"
  >
    Apple
  </li>
  <li
    className="breadcrumb__element"
  >
    Macbook
  </li>
  <li
    className="breadcrumb__element"
  >
    "
    macb
    "
  </li>
</ul>
`;
